File: clients/node/sdk/models/mediaAttachmentMetaOriginal.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MediaAttachmentMetaOriginal = void 0;
class MediaAttachmentMetaOriginal {
    /**
     * Gets the additionalData property value. Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
     * @returns a Record<string, unknown>
     */
    get additionalData() {
        return this._additionalData;
    }
    ;
    /**
     * Sets the additionalData property value. Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
     * @param value Value to set for the AdditionalData property.
     */
    set additionalData(value) {
        this._additionalData = value;
    }
    ;
    /**
     * Gets the aspect property value. The aspect property
     * @returns a float
     */
    get aspect() {
        return this._aspect;
    }
    ;
    /**
     * Sets the aspect property value. The aspect property
     * @param value Value to set for the aspect property.
     */
    set aspect(value) {
        this._aspect = value;
    }
    ;
    /**
     * Instantiates a new MediaAttachmentMetaOriginal and sets the default values.
     */
    constructor() {
        this._additionalData = {};
    }
    ;
    /**
     * The deserialization information for the current model
     * @returns a Record<string, (node: ParseNode) => void>
     */
    getFieldDeserializers() {
        return {
            "aspect": n => { this.aspect = n.getNumberValue(); },
            "height": n => { this.height = n.getNumberValue(); },
            "size": n => { this.size = n.getStringValue(); },
            "width": n => { this.width = n.getNumberValue(); },
        };
    }
    ;
    /**
     * Gets the height property value. The height property
     * @returns a integer
     */
    get height() {
        return this._height;
    }
    ;
    /**
     * Sets the height property value. The height property
     * @param value Value to set for the height property.
     */
    set height(value) {
        this._height = value;
    }
    ;
    /**
     * Serializes information the current object
     * @param writer Serialization writer to use to serialize this model
     */
    serialize(writer) {
        if (!writer)
            throw new Error("writer cannot be undefined");
        writer.writeNumberValue("aspect", this.aspect);
        writer.writeNumberValue("height", this.height);
        writer.writeStringValue("size", this.size);
        writer.writeNumberValue("width", this.width);
        writer.writeAdditionalData(this.additionalData);
    }
    ;
    /**
     * Gets the size property value. The size property
     * @returns a string
     */
    get size() {
        return this._size;
    }
    ;
    /**
     * Sets the size property value. The size property
     * @param value Value to set for the size property.
     */
    set size(value) {
        this._size = value;
    }
    ;
    /**
     * Gets the width property value. The width property
     * @returns a integer
     */
    get width() {
        return this._width;
    }
    ;
    /**
     * Sets the width property value. The width property
     * @param value Value to set for the width property.
     */
    set width(value) {
        this._width = value;
    }
    ;
}
exports.MediaAttachmentMetaOriginal = MediaAttachmentMetaOriginal;

File: clients/csharp/sdk/Models/MediaAttachmentMetaOriginal.cs
using Microsoft.Kiota.Abstractions.Serialization;
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
namespace MastodonClientLib.Models {
    public class MediaAttachmentMetaOriginal : IAdditionalDataHolder, IParsable {
        /// <summary>Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.</summary>
        public IDictionary<string, object> AdditionalData { get; set; }
        /// <summary>The aspect property</summary>
        public float? Aspect { get; set; }
        /// <summary>The height property</summary>
        public int? Height { get; set; }
        /// <summary>The size property</summary>
#if NETSTANDARD2_1_OR_GREATER || NETCOREAPP3_1_OR_GREATER
#nullable enable
        public string? Size { get; set; }
#nullable restore
#else
        public string Size { get; set; }
#endif
        /// <summary>The width property</summary>
        public int? Width { get; set; }
        /// <summary>
        /// Instantiates a new MediaAttachmentMetaOriginal and sets the default values.
        /// </summary>
        public MediaAttachmentMetaOriginal() {
            AdditionalData = new Dictionary<string, object>();
        }
        /// <summary>
        /// Creates a new instance of the appropriate class based on discriminator value
        /// </summary>
        /// <param name="parseNode">The parse node to use to read the discriminator value and create the object</param>
        public static MediaAttachmentMetaOriginal CreateFromDiscriminatorValue(IParseNode parseNode) {
            _ = parseNode ?? throw new ArgumentNullException(nameof(parseNode));
            return new MediaAttachmentMetaOriginal();
        }
        /// <summary>
        /// The deserialization information for the current model
        /// </summary>
        public IDictionary<string, Action<IParseNode>> GetFieldDeserializers() {
            return new Dictionary<string, Action<IParseNode>> {
                {"aspect", n => { Aspect = n.GetFloatValue(); } },
                {"height", n => { Height = n.GetIntValue(); } },
                {"size", n => { Size = n.GetStringValue(); } },
                {"width", n => { Width = n.GetIntValue(); } },
            };
        }
        /// <summary>
        /// Serializes information the current object
        /// </summary>
        /// <param name="writer">Serialization writer to use to serialize this model</param>
        public void Serialize(ISerializationWriter writer) {
            _ = writer ?? throw new ArgumentNullException(nameof(writer));
            writer.WriteFloatValue("aspect", Aspect);
            writer.WriteIntValue("height", Height);
            writer.WriteStringValue("size", Size);
            writer.WriteIntValue("width", Width);
            writer.WriteAdditionalData(AdditionalData);
        }
    }
}
